### PR TITLE
powerdns provider

### PR DESCRIPTION
**Pull request by jgreat**
This PR is to add a provider for powerdns (https://www.powerdns.com/) for an internally hosted dns solution.

We have a large amount of environments and none of the hosted providers could handle the api request rate.

I apologize upfront for the changes to all of the dependencies. I could not figure out how to resolve issues with the existing dependencies when trying to add github.com/jgreat/powerdns.  I will gladly work with someone to refactor this PR to make less invasive. 

The important bits are the following files:
providers/powerdns.go
metadata/metadata.go (only changed rancher metadata version because of upgrade to latest go-rancher-metadata)

### CODE REVIEW

**Comment by alena1108** [comment]
@jgreat thank you for the contribution! Have been busy working on Rancher 1.1 release, will try to review the PR this/early next week

**Comment by alena1108** [comment]
@janeczku could you give a first review to this PR please?

**Comment by alena1108** [comment]
@jgreat could you split your PR into 2 commits: one with godeps, and another with code changes?

**Comment by jgreat** [comment]
I took another crack at it and read up a bit more on godeps.  I think this is much cleaner and still works in my tests.  Please let me know if there is anything else you need.

File: providers/powerdns.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	var records []dns.DnsRecord
+
+	pdnsRecords, err := d.client.GetRecords()
+	if err != nil {
+		return records, fmt.Errorf("PowerDNS API call has failed: %v", err)
+	}
+
+	for _, rec := range pdnsRecords {
+		logrus.Debugf("%v\n", rec)
+		if rec.Type != "A" {
```
Comment (janeczku) [inline providers/powerdns.go]: We should keep it consistent with the existing providers and return any record type. Also Rancher will eventually support IPv6.

**Comment by janeczku** [comment]
Other than that LGTM. @jgreat Great PR!

**Comment by jgreat** [comment]
Removed the "only A records" bit.  Thanks for the feedback.

**Comment by alena1108** [comment]
LGTM

**Comment by alena1108** [comment]
@janeczku @jgreat do we have a corresponding rancher-catalog PR?

**Comment by jgreat** [comment]
@alena1108 I'll work up a pr for the catalog. Should that be for https://github.com/rancher/community-catalog

It doesn't look like there is a new docker build up yet. What will the image tag be? rancher/external-dns:v0.4.2

**Comment by alena1108** [comment]
@jgreat 0.5.0. We usually increment minor release when a new provider gets introduced, and patch release for bug fixes.

**Comment by fuhbar** [comment]
Great work, thank you @jgreat

**Pull request merged**